hw/ip/pinmux/data/pinmux.hjson.tpl

```diff
@@ -398,20 +398,53 @@
                   regwen:       "MIO_PAD_ATTR_REGWEN",
                   regwen_multi: "true",
                   cname:        "MIO_PAD",
+                  resval:       0
                   fields: [
-                    { bits: "12:0",
-                      name: "ATTR",
-                      desc: '''Bit   0:  input/output inversion,
-                               Bit   1:  Virtual open drain enable.
-                               Bit   2:  Pull enable.
-                               Bit   3:  Pull select (0: pull down, 1: pull up).
-                               Bit   4:  Keeper enable.
-                               Bit   5:  Schmitt trigger enable.
-                               Bit   6:  Open drain enable
-                               Bit 7/8:  Slew rate (0x0: slowest, 0x3: fastest).
-                               Bit 9/12: Drive strength (0x0: weakest, 0xf: strongest).
-                      '''
-                      resval: 0
+                    { bits: "0",
+                      name: "invert",
+                      desc: "Invert input and output levels."
+                    },
+                    { bits: "1",
+                      name: "virtual_od_en",
+                      desc: "Enable virtual open drain."
+                    },
+                    { bits: "2",
+                      name: "pull_en",
+                      desc: "Enable pull-up or pull-down resistor."
+                    },
+                    { bits: "3",
+                      name: "pull_select",
+                      desc: "Pull select (0: pull-down, 1: pull-up)."
+                      enum: [
+                        { value: "0",
+                          name:  "pull_down",
+                          desc:  "Select the pull-down resistor."
+                        },
+                        { value: "1",
+                          name:  "pull_up",
+                          desc:  "Select the pull-up resistor."
+                        }
+                      ]
```

Comment:
I guess this turned out to be useless, since enums appear to only generate macros in the C header if they are larger than 1 bit.

Reply:
This should be easy to fix. We can probably just remove the `field_width` qualification on this piece of code here:
https://github.com/lowRISC/opentitan/blob/1d750d8dc3af404939b58a565272e44a82d79ab7/util/reggen/gen_cheader.py#L160-L168
Same goes for the Rust header generation:
https://github.com/lowRISC/opentitan/blob/1d750d8dc3af404939b58a565272e44a82d79ab7/util/reggen/gen_rust.py#L146-L155

Reply:
Heh, nice, just need to delete 4 spaces from several lines. But I'll hold off on this for now, since the type technically is different from what we expect as an argument to `bitfield_bit32_write()` and a return value for `bitfield_bit32_read()`, though the conversion to bool is trivial.

It turns out I also didn't use the macro in the DIF. When that PR goes up, I guess we'll see if others think I should. :laughing: